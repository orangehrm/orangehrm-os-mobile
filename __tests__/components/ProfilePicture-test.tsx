/*
 * This file is part of OrangeHRM
 *
 * Copyright (C) 2020 onwards OrangeHRM (https://www.orangehrm.com/)
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import 'react-native';
import React from 'react';
import ProfilePicture from 'components/ProfilePicture';
import {Provider} from 'react-redux';
import configureStore from 'store/configureStore';
import {render} from 'react-native-testing-library';
const mockStore = configureStore();

describe('components/ProfilePicture', () => {
  test('test ProfilePicture component', () => {
    const node = render(
      <Provider store={mockStore}>
        <ProfilePicture
          employee={{
            empNumber: 1,
            employeeId: '0001',
            firstName: 'Abbey',
            lastName: 'Kyla',
            middleName: '',
            jobTitle: {id: 2, title: 'Accountant', isDeleted: false},
            subunit: {id: null, name: null},
            terminationId: null,
          }}
        />
      </Provider>,
    ).toJSON();
    expect(node).toMatchSnapshot();
  });

  test('test ProfilePicture component: terminated employee', () => {
    const node = render(
      <Provider store={mockStore}>
        <ProfilePicture
          employee={{
            empNumber: 10,
            employeeId: '0005',
            firstName: 'Lynda',
            lastName: 'Jane',
            middleName: '',
            jobTitle: {id: 6, title: 'Manager', isDeleted: false},
            subunit: {id: null, name: null},
            terminationId: 2,
          }}
        />
      </Provider>,
    ).toJSON();
    expect(node).toMatchSnapshot();
  });
});
