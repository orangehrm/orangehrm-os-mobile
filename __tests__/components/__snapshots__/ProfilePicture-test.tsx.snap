// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ProfilePicture test ProfilePicture component 1`] = `
<View>
  <View
    accessibilityIgnoresInvertColors={true}
    style={{}}
  >
    <Image
      defaultSource={
        {
          "testUri": "../../../src/images/default-photo.png",
        }
      }
      source={
        {
          "testUri": "../../../src/images/default-photo.png",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          {
            "opacity": 0.5,
            "resizeMode": "cover",
          },
        ]
      }
    />
    <View>
      <View
        style={
          {
            "elevation": 4,
          }
        }
      >
        <Image
          defaultSource={
            {
              "testUri": "../../../src/images/default-photo.png",
            }
          }
          large={true}
          source={
            {
              "testUri": "../../../src/images/default-photo.png",
            }
          }
          style={
            [
              {
                "borderRadius": 40,
                "height": 80,
                "width": 80,
              },
              {
                "margin": 16,
              },
            ]
          }
        />
      </View>
      <View
        style={
          {
            "marginBottom": 16,
            "marginLeft": 16,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#818181",
              },
              undefined,
              [
                {
                  "fontWeight": "bold",
                },
                {
                  "color": "#000000",
                },
              ],
            ]
          }
        >
          Abbey Kyla
        </Text>
        <Text
          style={
            [
              {
                "color": "#818181",
              },
              undefined,
              {
                "color": "#000000",
              },
            ]
          }
        >
          Accountant
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`components/ProfilePicture test ProfilePicture component: terminated employee 1`] = `
<View>
  <View
    accessibilityIgnoresInvertColors={true}
    style={{}}
  >
    <Image
      defaultSource={
        {
          "testUri": "../../../src/images/default-photo.png",
        }
      }
      source={
        {
          "testUri": "../../../src/images/default-photo.png",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          {
            "opacity": 0.5,
            "resizeMode": "cover",
          },
        ]
      }
    />
    <View>
      <View
        style={
          {
            "elevation": 4,
          }
        }
      >
        <Image
          defaultSource={
            {
              "testUri": "../../../src/images/default-photo.png",
            }
          }
          large={true}
          source={
            {
              "testUri": "../../../src/images/default-photo.png",
            }
          }
          style={
            [
              {
                "borderRadius": 40,
                "height": 80,
                "width": 80,
              },
              {
                "margin": 16,
              },
            ]
          }
        />
      </View>
      <View
        style={
          {
            "marginBottom": 16,
            "marginLeft": 16,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#818181",
              },
              undefined,
              [
                {
                  "fontWeight": "bold",
                },
                {
                  "color": "#000000",
                },
              ],
            ]
          }
        >
          Lynda Jane (Past Employee)
        </Text>
        <Text
          style={
            [
              {
                "color": "#818181",
              },
              undefined,
              {
                "color": "#000000",
              },
            ]
          }
        >
          Manager
        </Text>
      </View>
    </View>
  </View>
</View>
`;
